### PR TITLE
Adding ed25519 parallel signer

### PR DESCRIPTION
**Pull request by cviecco**

### CODE REVIEW

File: cmd/keymasterd/config.go [hunk 1]
```diff
@@ -268,6 +271,83 @@ func warnInsecureConfiguration(state *RuntimeState) {
...
+// or starts the autounselaing if encrypted
+func (state *RuntimeState) tryLoadAndVerifySigners() error {
+	state.logger.Debugf(2, "Top of tryLoadAndVerifySigners")
+	signerBlock, _ := pem.Decode(state.SSHCARawFileContent)
+	if signerBlock == nil {
+		// it is not PEM.. probably armor.. ie encrypted?
+		decbuf := bytes.NewBuffer(state.SSHCARawFileContent)
+		armorSignerBlock, err := armor.Decode(decbuf)
+		if err != nil {
+			return fmt.Errorf("signer content is not pem encoded or armor decocable")
```
Comment (rgooch) [inline cmd/keymasterd/config.go:322]: s/decocable/decodable/

File: cmd/keymasterd/unseal.go [hunk 2]
```diff
@@ -144,27 +138,35 @@ func (state *RuntimeState) unsealCA(password []byte, clientName string) error {
...
-	signer, err := getSignerFromPEMBytes(plaintextBytes)
+	signerPlaintextBytes, err := pgpDecryptFileData(state.SSHCARawFileContent, password)
 	if err != nil {
-		fmt.Errorf("cannot parse Private Key file: %s", err)
+		return err
 	}
-	logger.Printf("About to generate CA DER %s", clientName)
-	state.caCertDer, err = generateCADer(state, signer)
-	if err != nil {
-		return fmt.Errorf("cannot generate CA DER: %s", err)
```
Comment (rgooch) [inline cmd/keymasterd/unseal.go:160]: Error is ignored?

File: cmd/keymasterd/unseal.go [hunk 3]
```diff
@@ -144,27 +138,35 @@ func (state *RuntimeState) unsealCA(password []byte, clientName string) error {
...
-	plaintextBytes, err := ioutil.ReadAll(md.UnverifiedBody)
-	if err != nil {
-		return err
+	return ioutil.ReadAll(md.UnverifiedBody)
+}
+
+func (state *RuntimeState) unsealCA(password []byte, clientName string) error {
+	state.Mutex.Lock()
+	defer state.Mutex.Unlock()
+	// TODO.. make network error blocks to goroutines
```
Comment (rgooch) [inline cmd/keymasterd/unseal.go:149]: s/make/move/ ?

File: cmd/keymasterd/unseal.go [hunk 4]
```diff
@@ -144,27 +138,35 @@ func (state *RuntimeState) unsealCA(password []byte, clientName string) error {
...
+		return err
 	}
-	logger.Printf("About to generate CA DER %s", clientName)
-	state.caCertDer, err = generateCADer(state, signer)
-	if err != nil {
-		return fmt.Errorf("cannot generate CA DER: %s", err)
+	var ed25519PlaintextBytes []byte
+	if state.Ed25519CAFileContent != nil && len(state.Ed25519CAFileContent) > 0 {
+		ed25519PlaintextBytes, err = pgpDecryptFileData(state.Ed25519CAFileContent, password)
 	}
```
Comment (rgooch) [inline cmd/keymasterd/unseal.go:160]: Error is ignored?

**Review by rgooch — CHANGES_REQUESTED** [review]

File: cmd/keymasterd/unseal.go (hunk 4 above)
Comment (cviecco) [reply cmd/keymasterd/unseal.go:164]: fixed

File: cmd/keymasterd/config.go (hunk 1 above)
Comment (rgooch) [reply cmd/keymasterd/config.go:322]: I don't see this fixed?

**Pull request merged**